src/mappings/colony.ts

```diff
@@ -16,10 +29,37 @@ export function handleDomainAdded(event: DomainAdded): void {
   // event.transaction.input.toHexString()
   // And extract the parent that way. But it causes a memory-access out-of-bounds error which
   // isn't really a good sign...
-  domain.parent = event.address.toHex() + '_1'
+  domain.parent = event.address.toHex() + '_domain_1'
   domain.name = "Domain #" + event.params.domainId.toString()
+  if (event.params.domainId.toString() == '1') {
+    domain.name = "Root"
+    domain.parent = null
+  }
   domain.colonyAddress = event.address.toHex()
   domain.save()
+
+  handleEvent('DomainAdded(address,uint256)', event, event.address);
+}
+
+export function handleDomainMetadata(event: DomainMetadata): void {
+  let metadata = event.params.metadata.toString()
+  let domain = new Domain(event.address.toHex() + '_domain_' + event.params.domainId.toString())
+  domain.metadata = metadata
+
+  let metadataHistory = new DomainMetadataInstance(
+    event.address.toHex() +
+    '_domain_' + event.params.domainId.toString() +
+    '_transaction_' + event.transaction.hash.toHexString() +
+    '_log_' + event.logIndex.toString(),
```

Comment:
We should put the string 'metadata' in here somewhere - we could easily use the same logic for something that wasn't metadata and accidentally end up with the same ID. My fault though, you just did what I told you :sweat_smile:

Reply:
Done